File: Уровень2/Уровень2А/Задачи_на_префиксные_суммы/Сумма_на_отрезке.py
#https://algoprog.ru/material/p2771

n, m = map(int, input().split())
array = [*map(int, input().split())]

prefix = [0] * (n + 1)
for i in range(1, n+1):
    prefix[i] = prefix[i-1] + array[i-1]

for i in range(m):
    i, j = map(int, input().split())
    print(prefix[j] - prefix[i-1])

File: Уровень2/Уровень2А/Задачи_на_префиксные_суммы/Сумма_на_отрезке.cpp
#https://algoprog.ru/material/p2771

#include <iostream>
#include <vector>
using namespace std;

int main()
{
    int n, m;
    cin >> n >> m;
    vector<int> array;
    for (auto i = 0; i < n; i++) {
        int t;
        cin >> t;
        array.push_back(t);
    }
    vector<int> prefix;
    prefix.push_back(0);
    for (auto i = 1; i < n + 1; i++) {
        prefix.push_back(prefix[i-1] + array[i-1]);
    }
    for (auto k = 0; k < m; k++) {
        int i, j;
        cin >> i >> j;
        cout << prefix[j] - prefix[i-1] << endl;
    }
    return 0;
}
